blocks/block-edit/context.js

```diff
@@ -16,11 +16,36 @@ const { Consumer, Provider } = createContext( {
  * @return {Component} Component with a BlockEdit context set.
  */
 export const withBlockEditContextProvider = createHigherOrderComponent( ( OriginalComponent ) => {
-	return ( props ) => (
-		<Provider value={ { isSelected: props.isSelected } }>
-			<OriginalComponent { ...props } />
-		</Provider>
-	);
+	return class ComponentWithProvider extends Component {
```

Comment:
See https://reactjs.org/docs/context.html#caveats for more details.

> Because context uses reference identity to determine when to re-render, there are some gotchas that could trigger unintentional renders in consumers when a provider’s parent re-renders.